src/lib/PPrint.hs

```diff
@@ -336,7 +336,10 @@ instance Pretty (PiType n) where
     prettyBody = case body of
       Pi subpi -> pretty subpi
       _ -> pLowest body
-    in prettyBinder <> (group $ line <> p arr <+> prettyBody)
+    prettyEff = case eff of
+      Pure -> space
+      _    -> pretty eff <> space
```

Comment:
Should this be `space <> pretty eff <> space`, or do we know that `pretty eff` always starts with enough space?

Reply:
Good point! Fixed.